src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -123,11 +124,13 @@
 
         CityHash128Key(String sql,
                 String parametersDefinition) {
-            this(String.format("%s%s", sql, parametersDefinition));
+            this(sql + parametersDefinition);
         }
 
         CityHash128Key(String s) {
-            segments = CityHash.cityHash128(s.getBytes(), 0, s.length());
+            byte[] bytes = new byte[s.length()];
+            s.getBytes(0, s.length(), bytes, 0);
+            segments = CityHash.cityHash128(bytes, 0, bytes.length);
```

Comment:
@RDearnaley I would suggest going ahead with the slightly safer:
```
byte[] bytes = new byte[database.length() * 2];
for (int i = 0; i < database.length(); i++) {
   final int c = database.charAt(i);
   bytes[i * 2] = (byte) c;
   bytes[(i * 2) + 1] = (byte)(c >> 8);
}
segments = CityHash.cityHash128(bytes, 0, bytes.length);
```
<sup><sup>1</sup> Receiving the ``charAt()`` into an ``int`` avoids one implicit conversion, as shift operators are not supported on ``char`` directly and the ``(byte) c`` cast also likely inflates to an ``int`` before casting.</sup>